gpa.py

```diff
@@ -1,215 +1,148 @@
+   return grade, points
 
-   Parameters
-   -----------
-   grades : Array of grade 
-
-   Returns
-   -------
-   array_like ,  Array of points mapped to alphabets 5,4,3,2,1, 0.
-
-   Example
-   --------
-   grade=[60, 80, 40,]
-   point=[4,5,2]
-
-   """
-   grade_point_map = {'A': 5, 'B': 4, 'C':3, 'D':2, 'E':1, 'F':0}
-   point=[grade_point_map[grade] for grade in grades if grade in grade_point_map.keys()]
-   return point
+# print(compute_student_grade([50,60,84]))
+   
 
 # REVIEW COMMENTS
    # this function is not efficient
    # it should take a single semester data at a time 
    # the calling function should use a loop to call for each semester in turn
 # END REVIEW COMMENTS
-def compute_student_gpa(fall_credit, fall_unit_score, spring_credit, spring_unit_score):
+def compute_student_gpa(units, scores):
    """
-   Use the score obtained in each semester(fall or spring) and the course unit credit to 
-   calculates the gpa for the respective semester.
-
+   Use the score obtained in each semester(fall or spring) and the course unit credit to calculates the gpa for the respective semester.
    Parameters
    -----------
-   fall_credit : Array of int 
-   fall_unit_score : Array of int
-   spring_credit : array of int
-   spring_unit_score : array of int
-
+   units : Array of int 
+   scores : Array of int
    Returns
    -------
    Tuple of two float values
-
    Example
    --------
    compute_student_gpa([4, 5], [8, 15], [2,3], [1,2])) # (2.56, 0.6)
-
    """
-   fall_gpa= round((sum(fall_unit_score) / sum(fall_credit)), 2)
-   spring_gpa= round((sum(spring_unit_score) / sum(spring_credit)), 2)
-   return fall_gpa, spring_gpa
-
+   grade,points=compute_student_grade(scores)
+   total_unit=sum(units)
+   total_grade_point = sum([a*b for a,b in zip(units,points)])# 11 [a,a,e][5,5,1][3,4,2]
+   gpa= round((total_grade_point / total_unit), 2)
 
+   return total_unit,total_grade_point,gpa
 
-# REVIEW COMMENTS
-   # this function is not used anywhere 
-   # suggest to remove 
-# END REVIEW COMMENTS
-def output_box():
+def get_score_data(term,course_records):
    """
-   prints a hallow box for diplaying output
-
-   Parameters
-   ----------
-   row > int
-   columns > int
-
+   This function takes the term and student record and extracts the scores and the units for the semester.
+   parameters
+   -----------
+   string and dict
+   return
+   -------
+   array of units and scores
    """
-   rows = int(5)
-   columns = int(10)
-   for i in range(0, rows):
-      for j in range(0, columns):
-         if(i == 0 or i == rows - 1 or j == 0 or j == columns - 1):
-               print('#',  end = '  ')
-         else:
-               print(' ', end = '  ')
-      print()
-
-
-def compute_cgpa(student_record):
+   units=[]
+   scores=[]
+   for record in (course_records['courses']):
+      if record['term'] == term:
+         units.append(record['unit'])
+         scores.append(record['score'])
```

Comment:
This whole block could probably be rewritten more efficiently as: 
```suggestion
   units= [record['unit'] for record in course_records['courses'] if record['term'] == term)]
   scores= [record['score'] for record in course_records['courses'] if record['term'] == term)]
```